flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGeneratorContext.scala

```diff
@@ -505,8 +598,27 @@ class CodeGeneratorContext(val tableConfig: TableConfig) {
   }
 
   /**
-    * Adds a reusable local time to the beginning of the SAM of the generated class.
-    */
+   * Adds a reusable session-level local date time to the beginning of
+   * the SAM of the generated class.
+   */
+  def addReusableSessionLocalDateTime(): String = {
```

Comment:
Are you sure that this is enough? A session-level local time should be the same for multiple calcs in the pipeline. Ideally, across parallel instances. As far as I can see this would only ensure per-calc operator constant time here. We should find a solution that has better semantics.

Reply:
I could imagine that we have to persist the session time when generating the plan and making this available planner-wide.

Reply:
You're right, @twalthr , we need to consider the session time should be used in the entire plan.